src/EnergyPlus/HVACFourPipeBeam.hh

```diff
@@ -0,0 +1,280 @@
+		mDotDesignCW( 0.0 ),
+		qDotNormRatedCooling( 0.0 ),
+		deltaTempRatedCooling( 0.0 ),
+		vDotNormRatedCW( 0.0 ),
+		mDotNormRatedCW( 0.0 ),
+		modCoolingQdotDeltaTFuncNum( 0 ),
+		modCoolingQdotAirFlowFuncNum( 0 ),
+		modCoolingQdotCWFlowFuncNum( 0 ),
+		mDotCW( 0.0 ),
+		cWTempIn( 0.0 ),
+		cWTempOut( 0.0 ),
+		cWTempOutErrorCount( 0 ),
+		cWInNodeNum( 0 ),
+		cWOutNodeNum( 0 ),
+		cWLocation( PlantLocation( 0, 0, 0, 0 ) ),
+		beamHeatingPresent( false ),
+		vDotDesignHW( 0.0 ),
+		vDotDesignHWWasAutosized( false ),
+		mDotDesignHW( 0.0 ),
+		qDotNormRatedHeating( 0.0 ),
+		deltaTempRatedHeating( 0.0 ),
+		vDotNormRatedHW( 0.0 ),
+		mDotNormRatedHW( 0.0 ),
+		modHeatingQdotDeltaTFuncNum( 0 ),
+		modHeatingQdotAirFlowFuncNum( 0 ),
+		modHeatingQdotHWFlowFuncNum( 0 ),
+		mDotHW( 0.0 ),
+		hWTempIn( 0.0 ),
+		hWTempOut( 0.0 ),
+		hWTempOutErrorCount( 0 ),
+		hWInNodeNum( 0 ),
+		hWOutNodeNum( 0 ),
+		hWLocation( PlantLocation( 0, 0, 0, 0 ) ),
+		beamCoolingEnergy( 0.0 ),
+		beamCoolingRate( 0.0 ),
+		beamHeatingEnergy( 0.0 ),
+		beamHeatingRate( 0.0 ),
+		supAirCoolingEnergy( 0.0 ),
+		supAirCoolingRate( 0.0 ),
+		supAirHeatingEnergy( 0.0 ),
+		supAirHeatingRate( 0.0 ),
+		primAirFlow( 0.0 ),
+		myEnvrnFlag( true ),
+		mySizeFlag( true ),
+		plantLoopScanFlag( true ),
+		zoneEquipmentListChecked( false ),
+		tDBZoneAirTemp( 0.0 ),
+		tDBSystemAir( 0.0 ),
+		mDotSystemAir( 0.0 ),
+		cpZoneAir( 0.0 ),
+		cpSystemAir( 0.0 ),
+		qDotSystemAir( 0.0 ),
+		qDotBeamCoolingMax( 0.0 ),
+		qDotBeamHeatingMax( 0.0 ),
+		qDotTotalDelivered( 0.0 ),
+		qDotBeamCooling( 0.0 ),
+		qDotBeamHeating( 0.0 ),
+		qDotZoneReq( 0.0 ),
+		qDotBeamReq( 0.0 ),
+		qDotZoneToHeatSetPt( 0.0 ),
+		qDotZoneToCoolSetPt( 0.0 )
+	{}
+
+	// Destructor
+	virtual
+	~HVACFourPipeBeam()
+	{}
+
+public: // Methods		MARK ANY THAT DON'T ALTER STATE const !!!
+
+	// Needed for unit tests, should not be normally called.
+	void
+	clear_state();
+
+	static std::shared_ptr< AirTerminalUnit >
+	fourPipeBeamFactory(
+		int objectType,
+		std::string objectName
+	);
+
+	void
+	init(
+		bool const FirstHVACIteration // TRUE if first air loop solution this HVAC step         MAYBE THIS SHOULD HAVE A DEFAULT ARG OF = false
+	);
+
+	void
+	set_size();
+
+	Real64
+	residualSizing(
+		Real64 const airFlow // primary supply air flow rate in kg/s
+	);
+
+	void
+	control(
+		bool const FirstHVACIteration, // TRUE if 1st HVAC simulation of system timestep
+		Real64 & NonAirSysOutput // convective cooling by the beam system [W]
+	);
+
+	void
+	calc();
+
+	Real64
+	residualCooling(
+		Real64 const cWaterFlow // chilled water flow rate in kg/s
+	);
+
+	Real64
+	residualHeating(
+		Real64 const hWaterFlow // hot water flow rate in kg/s
+	);
+
+	void
+	simulate(
+		bool const FirstHVACIteration, // TRUE if first HVAC iteration in time step
+		Real64 & NonAirSysOutput // convective cooling by the beam system [W]
+	);
+
+	void
+	update() const;
+
+	void
+	report() ;
+
+public: // Data
```

Comment:
@EnergyArchmage Does this data need to be public? I haven't looked fully to see whether it does or not. Is it only used within this class or does it need to be accessed and used outside this class as well?